src/App.tsx

```diff
@@ -50,22 +51,102 @@ import ContentManager from "./api/ContentManager"
 import { Plugins, AppState } from '@capacitor/core';
 import NotificationsAssistant from "./api/NotificationsAssistant"
 
+
+
 // init content manager
 let cm = new ContentManager();
 
 const App: React.FC = () => {
 
-  // capacitor notifications setup
-  useIonViewDidEnter(() => {
-    // refresh notifications on app enter
-    Plugins.App.addListener("appStateChange", (state: AppState) => {
+
+  useEffect( () => {
+
+    console.log("Starting app")
+
+    let notificationAssistant = new NotificationsAssistant();
+
+    let remain:Boolean;
+    let starting:Number;
+    let ending:Number;
+
+    notificationAssistant.NoNotificationsRemaining().then((remaining) => {
+      if(!remaining){
+        console.log("There are notifications that are already scheduled");
+      }else{
+        console.log("There are notifications that aren't scheduled")
+         notificationAssistant.NotificationSetter();
+   
+      }
+      remain = remaining;
+    } ).then(()=>{
+
+       
+
+    })
+  })
+
+   
+
+           /*
+        Idea for the implementation
+
+        We need to check if there are any remaining notifications, either those that have not been received
+        Or those that have not been scheduled (if we take that route of allowing unlimited notifications)
+
+        If there are still scheduled notifications that have not been received, return out of the function (do nothing)
+        If there are notifications that need to be scheduled but are not yet scheduled, schedule any of them as long as the time has not been passed
+        
+        If no notifications, check to see if the start and end time has passed already
+          -If it has, do nothing
+          -If it has not passed, call the notification setter
+        */
+        
+   //Seems to not be called during startup
+   //https://stackoverflow.com/questions/50623279/js-event-handler-async-function/50623441
+   
+
+   /*Plugins.App.addListener("appStateChange", async (state: AppState) => {
       if (state.isActive) {
```

Comment:
Willie, why did you get rid of this code? Wasnt the whole point of the App.addListener to have the app refresh when you switch from background to foreground? This is a real problem, as major functionality is being commented out...

Reply:
So it's going to reload when we either open the app, or when we go from background to foreground? I think I might have removed that part at one point because it wasn't being called on app start up

Reply:
I think the fix is just to uncomment the app listener part, and just replace the block of code with the block of code in useEffect, which is called during the start up of the App

Reply:
Sounds like a plan. But using https://ionicframework.com/docs/react/lifecycle#lifecycle-methods-in-functional-components would be better, since it is only called once the page is started up.